exp/runtime/hooks/api/v1alpha1/lifecyclehooks_types.go

```diff
@@ -0,0 +1,308 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog"
+)
+
+// BeforeClusterCreateRequest is the request of the BeforeClusterCreate hook.
+// +kubebuilder:object:root=true
+type BeforeClusterCreateRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// The cluster object the lifecycle hook corresponds to.
+	Cluster clusterv1.Cluster `json:"cluster"`
+}
+
+var _ AggregatableResponse = &BeforeClusterCreateResponse{}
+
+// BeforeClusterCreateResponse is the response of BeforeClusterCreate hook.
+// +kubebuilder:object:root=true
+type BeforeClusterCreateResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Status of the call. One of "Success" or "Failure".
+	Status ResponseStatus `json:"status"`
+
+	// RetryAfterSeconds when set to a non-zero signifies that the hook
+	// will be called again at a future time.
+	RetryAfterSeconds int `json:"retryAfterSeconds"`
+
+	// A human-readable description of the status of the call.
+	Message string `json:"message"`
+}
+
+// Aggregate combines individual extension handler responses into a single response of the hook.
+func (r *BeforeClusterCreateResponse) Aggregate(responses []*ExtensionHandlerResponse) error {
+	if err := aggregateExtensionResponses(r, responses); err != nil {
+		return errors.Wrap(err, "failed to aggregate extension responses")
+	}
+	return nil
+}
+
+// BeforeClusterCreate is the runtime hook that will be called right before a Cluster is created.
+func BeforeClusterCreate(*BeforeClusterCreateRequest, *BeforeClusterCreateResponse) {}
+
+// AfterControlPlaneInitializedRequest is the request of the hook.
+// +kubebuilder:object:root=true
+type AfterControlPlaneInitializedRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// The cluster object the lifecycle hook corresponds to.
+	Cluster clusterv1.Cluster `json:"cluster"`
+}
+
+var _ AggregatableResponse = &AfterControlPlaneInitializedResponse{}
+
+// AfterControlPlaneInitializedResponse is the response of AfterControlPlaneInitialized hook.
+// +kubebuilder:object:root=true
+type AfterControlPlaneInitializedResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Status of the call. One of "Success" or "Failure".
+	Status ResponseStatus `json:"status"`
+
+	// A human-readable description of the status of the call.
+	Message string `json:"message"`
+}
+
+// Aggregate combines individual extension handler responses into a single response of the hook.
+func (r *AfterControlPlaneInitializedResponse) Aggregate(responses []*ExtensionHandlerResponse) error {
+	if err := aggregateExtensionResponses(r, responses); err != nil {
+		return errors.Wrap(err, "failed to aggregate extension responses")
+	}
+	return nil
+}
+
+// AfterControlPlaneInitialized is the runtime hook that will be called after the first control plane is available.
```

Comment:
```suggestion
// AfterControlPlaneInitialized is the runtime hook that will be called after the control plane is available.
```
or 
```suggestion
// AfterControlPlaneInitialized is the runtime hook that will be called after the control plane is available for the first time.
```
nit. I assume we don't have multiple control planes :)